internal/subscriber/subscriber.go

```diff
@@ -208,9 +167,8 @@ func (s *Subscriber) acknowledge(req *AckMessage) {
 
 		logger.Ctx(ctx).Infow("subscriber: msg hit deadline", "logFields", logFields)
 
-		// push to retry queue
-		s.retry(ctx, NewRetryMessage(msg.Topic, msg.Partition, msg.Offset, msg.Data, msgID, msg.RetryCount))
-
+		// push for retry
+		s.retry(ctx, msg)
 		s.removeMessageFromMemory(ctx, stats, req.MessageID)
```

Comment:
removing will mean that message is acknowledged as well, which is incorrect, its still being retried, in memory removal shouldn't happen until we move to DLQ